### PR TITLE
CB-15504: Validate the update-azure-encryption-resources request based on key provided

### PR DESCRIPTION
**Pull request by juhi-09**
CB-15504: Validate the update-azure-encryption-resources request based on key provided

Following changes are made as part of this commit -
 - If user tries to update the encryption key for an existing environment -
   1. If key doesn't exists - Update the encryption resources after validating the key.
   2. If same key already exists - Log info message and return the encryption resources without any error.
   3. If some key already exists - throw an error as only adding the key to existing environments is allowed and not modifying the key.

### CODE REVIEW

**Review by cegganesh84 — APPROVED** [review]
LGTM

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java [hunk 1]
```diff
@@ -177,26 +177,31 @@ private EnvironmentDto changeCredential(String accountId, String environmentName
...
+                LOGGER.debug("Successfully created the Disk encryption set for the environment {}.", environmentName);
+            } else {
+                throw new BadRequestException(validateKey.getFormattedErrors());
             }
-            LOGGER.debug("Successfully created the Disk encryption set for the environment {}.", environmentName);
+            return environmentDtoConverter.environmentToDto(environment);
         } else {
-            throw new BadRequestException(validateKey.getFormattedErrors());
+            throw new BadRequestException(String.format("Encryption Key '%s' is already set for the environment '%s'. " +
+                    "Modifying the encryption key is not allowed.", azureParameters.getEncryptionKeyUrl(), environmentName));
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:203]: What if the user specifies exactly the same key in the edit request that was used earlier to encrypt the environment? Is this possible, or rather there is a validation for that scenario in somewhere else? (Strictly speaking, that should not really count as an "edit" as no changes would be made to the encryption settings. Perhaps the request could be just ignored silently in that case, but I leave that to you to decide how to deal with this.)

**Review by lajosrodek — CHANGES_REQUESTED** [review]
The changes look mostly alright, but the commit summary & PR title are misleading. I was eagerly looking for the CLI additions, obviously to no avail as they were delivered in the thunderhead change https://github.infra.cloudera.com/thunderhead/thunderhead/pull/6842 . So please rephrase them to avoid confusions. It is surely okay to mention CB-15504 if you feel raising a separate ticket for this validation would not be worth the effort.

Apart from this the changes are fine; just had one q regarding the edge case when the same key is specified in the edit request as the one used for the env.

One more tiny note: Not a big deal but the phrase "This PR makes following changes" does not make much sense in git commit descriptions. GitHub will automatically add the PR ID tracking to the commit, so it becomes obvious whether the commit was merged as part of a PR or not (and if so, which PR ID). Saying "This commit makes following changes" sounds more appropriate in my opinion.

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java (hunk 1 above)
Comment (juhi-09) [reply environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:203]: I have updated the code to not throw any error if user tries to update encryption resources using the same encryption key, instead log the info message and return the encryption details as it is.

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Thanks for the "same key" case check & test.

The only remaining concern I have is with the commit title & PR title, both still mentioning `Append the DES details to response of update-azure-encryption-resources CLI`. This is misleading, the changes here are related to checking for the absence of an encryption key for the env (or the same key as the one passed in the request).

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

**Pull request merged**